gnovm/pkg/gnolang/preprocess.go

```diff
@@ -247,10 +246,18 @@ func initStaticBlocks(store Store, ctx BlockNode, bn BlockNode) {
 			case *RangeStmt:
 				if n.Op == DEFINE {
 					if n.Key != nil {
-						last.Predefine(false, n.Key.(*NameExpr).Name)
+						nx := n.Key.(*NameExpr)
+						if nx.Name != blankIdentifier {
+							nx.Type = NameExprTypeDefine
+							last.Predefine(false, nx.Name)
+						}
 					}
```

Comment:
maybe we should not support loopvar with range either, to make it consistent with for loop.

Reply:
yes. but TypeDefine is correct here as it is a define.